src/signal/interrupt.rs

```diff
@@ -48,17 +48,12 @@ impl Interrupt {
 
     pub fn send(changes: &[Change]) {
         let mut ch = SOURCE_CHANGES.blocking_write();
-        let mut did_change = false;
         for change in changes {
-            did_change |= ch.add(change.clone());
+            ch.add(change.clone());
```

Comment:
This change correctly notifies the watcher to terminate the running commands and rerun them.